### PR TITLE
Update controllers with new methods for handling relationships

### LINKED ISSUE #273: Rewrite relationships in controllers to make use of deserialization
You can see usage of this in `ProjectCategoriesController` in #272, or just after it's merged.

### PR DESCRIPTION
**Pull request by joshsmith**
First stab at #273. Closes #273 if merged.

### CODE REVIEW

**Comment by joshsmith** [comment]
I am open to any and all feedback here. I'm trying to make use of `ActiveModelSerializers::Deserialization.jsonapi_parse` which seems to cut down on a lot of the custom boilerplate we were writing in our own controllers.

I'm open to any feedback on whether this is a necessary change (although it seemed hard to do #272 without it), or if we need to improve on the syntax here. It's a pretty rough first pass.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -64,6 +64,22 @@ def render_error(error)
     render json: error_hash, status: error_hash[:errors][0][:status]
   end
 
+  def parse_params(params, options = {})
+    ActiveModelSerializers::Deserialization.jsonapi_parse(params, options)
+  end
+
+  def deserialized_params
+    parse_params(params)
```
Comment (beauby) [inline app/controllers/application_controller.rb]: Here you could return `ActionController::Parameters.new(parse_params(params))`.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -64,6 +64,22 @@ def render_error(error)
...
+  end
+
+  def require_param(key)
+    deserialized_params[key].presence || raise(ActionController::ParameterMissing.new(key))
```
Comment (beauby) [inline app/controllers/application_controller.rb]: And this would become `deserialized_params.require(key)`.

**Comment by JoshSmith** [comment]
Testing.

**Comment by begedin** [comment]
Looks good. I would just look into what @beauby is saying. Seems like it could be a bit cleaner.

**Comment by joshsmith** [comment]
Yeah, I don't like the `require_param` thing at all. Hopefully his Rails PR goes through.

**Pull request merged**